### DISCUSSION TITLE
Thoughts on new inproblem attempts feedback.

### DISCUSSION THREAD

**Discussion by somiaj** [openwebwork/webwork2#2240] [General]
I have finally got to test out the new inline feedback, and I like it a lot. Here are some initial impressions and thoughts I have.

First, I find that on some of my questions, showing the entered answer, preview answer, and correct answer vertically doesn't fit that nicely on a standard 16:9 monitor, and I would prefer to have the feedback in a horizontal row if the browser is wide enough. But I have some other ideas below.

In many cases (especially when using MathQuill) I don't see needing both the entered answer and previewed answer. Would there be a nice way to only show one of them.  Show the preview answer unless there is a parsing error, then revert to the entered answer? I know there might be some issues with this and MultiAnswer, so unsure if this could be done in a nice way.

What about moving the correct answer to its own popup? Add a second button with a key icon, that shows the correct answer.  This way the key icon can be shown whenever someone has permissions to see the answer, and there is no longer a need to have a 'show correct answer' button or checkbox.  If someone wants to see the correct answer, they just click the key icon.

Last, I also think we should remove the 'show solutions' checkbox (I find students don't really notice this or even realize I write solutions). If someone has permissions to see the solutions, the button which shows the solutions should just be visible, and someone can see them by just clicking, without reloading/checking their problem.

**Comment by drgrice1** [comment]
There has been an option to not show the "Entered" answer since before MathQuill was added to WeBWorK.  It is in the course configuration, and is the "Display the evaluated student answer" option in the "Problem Display/Answer Checking" tab.  Set that to false.

The feedback really won't work in a horizontal row, and should fit fine even with a 16:9 monitor.

Another button is a really bad idea, and another popup requires another button.  Many problems have multiple answer on the same line, and that would really mess things up.  In general adding multiple buttons will become intrusive and even confusing.

Removing the checkboxes is something that would be nice to do, and with this that is possible but with using the existing popup for correct answers.

When it comes to solutions, even if it is shown whenever someone has permissions, you may find that the students still don't know it is there since it is only shown after the due date.  I find that students rarely go back into assignments after the due date even though I tell them to.

**Comment by somiaj** [comment]
Thanks for the feedback. I was testing some of my multianswer problems with this, and due to the size of the entered/answer/correct answer boxes, the popup was bigger than a standard answer, and I was narrowly focused on the look for these specific problems. The vertical alignment works good for most problems.

Thanks for the reminder of the feature to turn off the entered column, forgot that was there. One thing I noticed is the help comment on that is now outdated, it states that "the student can still see their evaluated answer by clicking on the typeset version of their answer.", which is no longer correct since the popup with their evaluated answer is now gone. Should probably clean that up.

I find in my classes I can get many good students to go back and look at their feedback and questions, but they still miss there are solutions, so it will be nice to just not have that extra step to get to solutions/correct answers for the students that do go back to the assignments.

**Comment by drgrice1** [comment]
I assume you are using MultiAnswer with singleResult. With this new feedback mechanism, you may want to reevaluate using singleResult.  In many cases singleResult has been used to merely to make the results table not be so long.  That is no longer a valid reason to use singleResult with this approach at feedback.

Yes, the help comment on the evaluated answer option needs to be updated. Thanks for pointing that out.
